src/streamlink/plugins/vimeo.py

```diff
@@ -61,11 +65,45 @@ class Vimeo(Plugin):
         validate.any(None, validate.Schema(validate.get(1), _config_schema)),
     )
 
+    def get_config_url(self) -> Optional[str]:
+        viewer = self.session.http.get(
+            self.VIEWER_URL,
+            schema=validate.Schema(
+                validate.parse_json(),
+                {
+                    "jwt": str,
+                    "apiUrl": str,
+                },
+            ),
+        )
+
+        uri = self.session.http.get(
+            self.OEMBED_URL,
+            params={"url": self.url},
+            schema=validate.Schema(
+                validate.parse_json(),
+                {
+                    "uri": str,
+                },
+            ),
+        )
+
+        return self.session.http.get(
+            urljoin(update_scheme("https://", viewer["apiUrl"]), uri["uri"]),
+            params={"fields": "config_url"},
+            headers={"Authorization": "jwt {}".format(viewer["jwt"])},
+            schema=self._config_url_schema,
+        )
```

Comment:
That doesn't seem to be right. The API returns JSON data, so why is the `_config_url_schema` schema being applied here? Just parse the JSON payload and get the `config_url` key.

Some other minor improvements

```suggestion
        jwt, api_url = self.session.http.get(
            self.VIEWER_URL,
            schema=validate.Schema(
                validate.parse_json(),
                {
                    "jwt": str,
                    "apiUrl": str,
                },
                validate.union_get("jwt", "apiUrl"),
            ),
        )
        uri = self.session.http.get(
            self.OEMBED_URL,
            params={"url": self.url},
            schema=validate.Schema(
                validate.parse_json(),
                {"uri": str},
                validate.get("uri"),
            ),
        )
        player_config_url = urljoin(update_scheme("https://", api_url), uri)

        return self.session.http.get(
            player_config_url,
            params={"fields": "config_url"},
            headers={"Authorization": f"jwt {jwt}"},
            schema=validate.Schema(
                validate.parse_json(),
                {"config_url": validate.url()},
                validate.get("config_url"),
            ),
        )
```

The `Optional` typing annotation is also not correct because the return value never can be `None`. If an invalid video URL gets set by the user, then the `self.url` request returns a 404 error. An in case the API request doesn't include a proper config URL, this is guarded by the validation schema, which prints a meaningful error message on failure.